core/include/prometheus/histogram.h

```diff
@@ -67,8 +70,8 @@ class PROMETHEUS_CPP_CORE_EXPORT Histogram {
 
  private:
   const BucketBoundaries bucket_boundaries_;
-  std::vector<Counter> bucket_counts_;
-  Counter sum_;
+  std::atomic<double> sum_{0.0};
+  std::vector<std::atomic<double>> bucket_counts_;
```

Comment:
Why was this modified?

Reply:
Previously histogram used a series of counters to keep track of it's buckets and counters used guages to keep track of their values. This nesting of data structures proved troublesome when we now want to keep track of retention time for each metric (we don't want each bucket to have it's own retention time, and each counter in each bucket have it's own retention time >_<)